Supertext.Base.Dal.SqlServer.Tests/Utils/SqlResultConverterTest.cs

```diff
@@ -165,4 +165,180 @@ public void DecodeStructure_JsonEncodedFieldNames_Level2Json()
+    {
+        var testData = new TestEntity(Guid.NewGuid(),
+                                      "Test",
+                                      DateTime.Now.AddDays(-2),
+                                      DateTime.Now);
+
+        var converter = new SqlResultConverter();
+        var results = converter.InterpretUtcDates(testData);
+
+        results.Id.Should().Be(testData.Id);
+        results.Name.Should().Be(testData.Name);
+        results.CreatedOn.Kind.Should().Be(DateTimeKind.Utc);
+        results.UpdatedOn!.Value.Kind.Should().Be(DateTimeKind.Utc);
+    }
+
+    [TestMethod]
+    public void InterpretUtcDates_ObjectWithNestedObject_ConvertedInUtc()
+    {
+        var nestedObject = new TestEntity(Guid.NewGuid(),
+                                          "Nested",
+                                          DateTime.Now,
+                                          null);
+
+        var testData = new TestEntity(Guid.NewGuid(),
+                                      "Test",
+                                      DateTime.Now.AddDays(-2),
+                                      DateTime.Now,
+                                      nestedObject);
+
+        var converter = new SqlResultConverter();
+        var results = converter.InterpretUtcDates(testData);
+
+        results.Id.Should().Be(testData.Id);
+        results.Name.Should().Be(testData.Name);
+        results.CreatedOn.Kind.Should().Be(DateTimeKind.Utc);
+        results.UpdatedOn!.Value.Kind.Should().Be(DateTimeKind.Utc);
+        results.NestedObject!.Id.Should().Be(nestedObject.Id);
+        results.NestedObject.Name.Should().Be(nestedObject.Name);
+        results.NestedObject.CreatedOn.Kind.Should().Be(DateTimeKind.Utc);
+    }
+
+    [TestMethod]
+    public void InterpretUtcDates_ObjectWithMultipleNestedObject_ConvertedInUtc()
+    {
+        var nestedObject1 = new TestEntity(Guid.NewGuid(),
+                                          "Nested 1",
+                                          DateTime.Now,
+                                          null);
+
+        var nestedObject2 = new TestEntity(Guid.NewGuid(),
+                                           "Nested 2",
+                                           DateTime.Now.AddDays(-5),
+                                           DateTime.Now,
+                                           nestedObject1);
+
+        var testData = new TestEntity(Guid.NewGuid(),
+                                      "Test",
+                                      DateTime.Now.AddDays(-2),
+                                      DateTime.Now,
+                                      nestedObject2);
+
+        var converter = new SqlResultConverter();
+        var results = converter.InterpretUtcDates(testData);
+
+        results.Id.Should().Be(testData.Id);
+        results.Name.Should().Be(testData.Name);
+        results.CreatedOn.Kind.Should().Be(DateTimeKind.Utc);

```

Comment:
Again, I don't think we should have assertions for values which were covered in preceding tests. This method exists to verify that multiple nested objects (_i.e._, `DateTime` instances two levels deep) get converted to `DateTimeKind.Utc`, and it should not be looking at anything else.